cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +794,56 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+const char* gdv_fn_concat_ws_utf8(int64_t context, const char* separator,
+                                  int32_t separator_len, const char* data,
+                                  int32_t data_len, int32_t* out_len) {
+  if (data_len <= 0) {
+    gdv_fn_context_set_error_msg(context, "Data can not be null.");
+    *out_len = 0;
+    return "";
+  }
+
+  std::string concat = "";
+  std::string data_str(data, data_len);
```

Comment:
Please do not use this pattern - this does a memcpy and allocates memory

We should allocate memory for the output and then iterate over all the input buffers and copy them/separator to the output buffer

Reply:
Fixed.